src/slurm_plugin/resume.py

```diff
@@ -115,6 +129,18 @@ def _handle_failed_nodes(node_list):
 
 def _resume(arg_nodes, resume_config):
     """Launch new EC2 nodes according to nodes requested by slurm."""
+    # Check heartbeat
+    current_time = datetime.now(tz=timezone.utc)
+    if not is_clustermgtd_heartbeat_valid(
+        current_time, resume_config.clustermgtd_timeout, resume_config.clustermgtd_heartbeat_file_path
+    ):
+        log.error(
+            "No valid clustermgtd heartbeat detected, clustermgtd is down!\n"
+            "Please check clustermgtd log for error.\n"
+            "Not launching nodes %s",
+            arg_nodes,
+        )
+        raise Exception("Clustermgtd down")
```

Comment:
nit: can we handle it here rather than raising and having this caught in the generic unhandled exception block of the main function?